File: 0219.ContainsDuplicateII/ContainsDuplicateII.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        record = {}
        for i, num in enumerate(nums):
            if num in record and k >= abs(record.get(num, i) - i):
                return True
            record[num] = i
        return False

File: 0219.ContainsDuplicateII/ContainsDuplicateII.go
package main

func ContainsNearbyDuplicate(nums []int, k int) bool {
    mapping := map[int]int{}
    for i, val := range nums {
        if _, ok := mapping[val]; ok && i - mapping[val] <= k {
            return true
        }
        mapping[val] = i
    }
    return false
}

func main() {

}
